fuzz/fuzz_targets/c_rust_merkle.rs

```diff
@@ -8,23 +8,29 @@ use simplicity::jet::Elements;
 use simplicity::{BitIter, RedeemNode};
 
 fn do_test(data: &[u8]) {
-    if data.len() < 4 {
-        return;
-    }
-    let prog_len = (usize::from(data[0]) << 24)
-        + (usize::from(data[1]) << 16)
-        + (usize::from(data[2]) << 8)
-        + usize::from(data[3]);
-    if prog_len >= data.len() {
-        return;
-    }
-    let data = &data[4..];
+    // To decode the program length, we first try decoding the program using
+    // `decode_expression` which will not error on a length check. Alternately
+    // we could decode using RedeemNode::decode and then extract the length
+    // from the error return.
+    //
+    // If the program doesn't decode, just use the first byte as a "length".
+    let prog_len = {
+        let mut iter = BitIter::from(data);
+        match simplicity::decode::decode_expression::<_, Elements>(&mut iter) {
+            Ok(_) => (iter.n_total_read() + 7) / 8,
+            Err(_) => match data.first() {
+                Some(&n) => core::cmp::min(data.len(), n.into()),
+                None => return,
+            },
+        }
+    };
```

Comment:
8d3eda887b7adaf8a890701e3625db0ac19b6672: Do we compute the length of the program from the fuzzer bytes to help the fuzzer, because most random byte sequences won't have a valid length prefix?